controllers/volumereplicationgroup_controller.go

```diff
@@ -146,6 +156,68 @@ func (r *VolumeReplicationGroupReconciler) configMapFun(configmap client.Object)
 	return req
 }
 
+func (r *VolumeReplicationGroupReconciler) waitForCrds() error {
+	requiredCrds := []string{"VolumeReplicationClass", "VolumeReplication"}
+
+	for _, crd := range requiredCrds {
+		err := r.waitForResource(crd)
+		if err != nil {
+			r.Log.Error(err, "unexpected error occurred while checking CRD", "CRD", crd)
+
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (r *VolumeReplicationGroupReconciler) waitForResource(resourceName string) error {
+	result, err := r.resourceExists(resourceName)
```

Comment:
We must handle the error first *before* checking the result, so:

    if err != nil {
        return err
    }

Reply:
Done